### PR TITLE
HomeAssistant 0.89.1

### PR DESCRIPTION
**Pull request by andr2000**
This is a compile only checked version of the HomeAssistant v0.89.1 against thud branch of poky/meta-openembedded.
I would love to start discussing this if @bachp's will

### CODE REVIEW

File: conf/layer.conf [hunk 1]
```diff
@@ -10,3 +10,5 @@ BBFILE_PRIORITY_homeassistant = "10"
 
 # Most dependencies come from the meta-python layer
 LAYERDEPENDS_homeassistant = "core meta-python"
+
+PREFERRED_VERSION_python3-pytz="2018.7"
```
Comment (bachp) [inline conf/layer.conf]: Why is this needed? I usually try to make has work with what is available in meta-python if newer.

**Comment by bachp** [comment]
@andr2000 Thanks a lot for this PR. It looks very good. I will give it a try to see if it also works against latest master instead of thud. As the intention is that the master branch is always tracking master.

File: conf/layer.conf (hunk 1 above)
Comment (andr2000) [reply conf/layer.conf]: will remove

Comment (andr2000) [reply conf/layer.conf]: By your comment

> I usually try to make has work with what is available in meta-python if newer.

do you mean if a newer version of a python package is required by HA and an older one already exists in meta-python you try to use the existing one?
If so, then this PR has pytz and requests packages of this sort, e.g. older versions of these packages exist in meta-python and newer are introduced by this PR.

Comment (bachp) [reply conf/layer.conf]: Sorry I meant the other way around. If a newer version, than required by HA, is available in meta-python, I try to use it. If it works I also try to submit a PR to HA to bump the version requirement there.

If HA requires a newer version then the one in meta-python I usually submit an update to meta-python and add it as a backport to this layer.

I hope this clarifies what I meant.

Comment (andr2000) [reply conf/layer.conf]: It is all clear now. As I have no HW I target for HA (and what is more I have never run HA before ;)
is it possible for you to test the new version if it runs ok?
Thank you in advance

Comment (bachp) [reply conf/layer.conf]: I'm also not running HA via OE anymore. That's also the case why I got around to update the layer.

May I ask why you are doing the layer update if you are not using HA this way? Just out of curiosity :smile:

Comment (andr2000) [reply conf/layer.conf]: Ah, it's all simple:) I have a hobby project and HA will be part of it. I am about to run it on a Raspberry Pi board which is on its way to me, so this PR is in preparation for its arrival ;)

So I'll wait for RPi then and try HA if it runs and if it needs more dependencies satisfied.
BTW, what was your way to detect all python deps for HA? Can it be automated somehow?

Comment (bachp) [reply conf/layer.conf]: If you are interested it would be helpful to have an additional maintainer that is actually using the layer.

I dit it mostly manually, bu I think we could hack up a script that does the work. I just never got around to do it.

Comment (andr2000) [reply conf/layer.conf]: Thank you for the offer, but I am not sure I'll be able to spend much time on this :(

**Comment by bachp** [comment]
Unfotrunately this doesn't build against yocto master. I don't think it's related to this PR but I need some more time to resolve the issue.

I'm currently stuck with `python3-ua-parser` not building.

**Comment by andr2000** [comment]
> Unfotrunately this doesn't build against yocto master. I don't think it's related to this PR but I need some more time to resolve the issue.
> 
> I'm currently stuck with `python3-ua-parser` not building.

I have received RPi and in the middle of bringing it up. HA is the highest priority after that, but still I am on thud

**Comment by bachp** [comment]
I will merge this as it and then I can fix the master issues on top of that. It doesn't build anyway.

**Comment by christiansandberg** [comment]
Did anyone try to make this work? In general there seem to be mismatches between [required versions](https://github.com/home-assistant/home-assistant/blob/0.89.1/requirements_all.txt) by homeassistant and versions provided in the layer.

**Comment by bachp** [comment]
@christiansandberg No I never checked it. I'm currently not using this anymore but I'm happy to find a new maintainer.

**Comment by christiansandberg** [comment]
I appreciate your efforts! I was mostly playing around with Yocto as some kind of kiosk but my plan was to have homeassistant run on a real distro later anyway and just use Yocto as an interface.

**Comment by bachp** [comment]
I found it easier to run homeassistant as a container using podman or docker instead of chasing dependencies every few weeks.

**Comment by andr2000** [comment]
I am still going to continue working on this later or sooner. Unfortunately I am busy with other things atm :(
